app/main.py

```diff
@@ -2,0 +1,30 @@
+class Animal:
+    alive = []
+
+    def __init__(
+            self,
+            name: str,
+            health: int = 100,
+            hidden: bool = False) -> None:
+        self.name = name
+        self.health = health
+        self.hidden = hidden
+        Animal.alive.append(self)
+
+    def __repr__(self) -> str:
+        return (f"{{Name: {self.name}, "
+                f"Health: {self.health}, "
+                f"Hidden: {self.hidden}}}")
+
+
+class Herbivore(Animal):
+    def hide(self) -> None:
+        self.hidden = not self.hidden
+
+
+class Carnivore(Animal):
+    def bite(self, animal: Herbivore) -> None:
+        if not animal.hidden and isinstance(animal, Herbivore):
+            animal.health -= 50
+            if animal.health <= 0:
+                self.alive.remove(animal)
```

Comment:
The `alive` list is accessed through `self.alive`, which implies it should be an instance attribute. However, it is currently a class attribute. If you intend to keep it as a class attribute, use `Animal.alive.remove(animal)` instead. Otherwise, consider changing `alive` to an instance attribute.